src/main/java/io/stargate/sgv2/jsonapi/service/cqldriver/executor/CollectionSchemaObject.java

```diff
@@ -26,27 +26,71 @@
 /**
  * Refactored as seperate class that represent a collection property.*
  *
- * @param collectionName
- * @param vectorConfig
- * @param indexingConfig
+ * <p>TODO: there are a LOT of different ways this is constructed, need to refactor
  */
-public record CollectionSettings(
-    String collectionName,
-    IdConfig idConfig,
-    VectorConfig vectorConfig,
-    IndexingConfig indexingConfig) {
-
-  private static final CollectionSettings EMPTY =
-      new CollectionSettings(
-          "", IdConfig.defaultIdConfig(), VectorConfig.notEnabledVectorConfig(), null);
-
-  public static CollectionSettings empty() {
-    return EMPTY;
+public final class CollectionSchemaObject extends SchemaObject {
```

Comment:
Noted it somewhere else, but thinking it might make sense to have an intermediate base SchemaObject type between `SchemaObject` and this class (shared base class with `TableSchemaObject`).

Reply:
Did not follow, why another intermediate class.

Reply:
For functionality applicable to "Collection OR Table" (but not Keyspace or Database), given that they are quite similar in many ways.
(may not make sense, might, just bringing up for consideration)

Reply:
Possible idea for near future, not changed in initial commits.